### PR TITLE
Update module

### PR DESCRIPTION
**Pull request by retyui**
-  [TS] no children error
-  [TS] `component` & `computeFont` are required props error
-  [Example] Add example
-  [Feat] Optimise style assignment (without spread)

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -1,52 +1,40 @@
...
-  return Component ? (
+  return (
     <Component
-      style={{ ...combinedStyle, fontFamily, fontWeight: undefined }}
-      {...props}
-    />
-  ) : (
-    <RNText
-      style={{ ...combinedStyle, fontFamily, fontWeight: undefined }}
+      style={[style, { fontFamily, fontWeight: undefined }]}
```
Comment (retyui) [inline src/index.tsx:37]: here not needed an expensive `spread` operation on each render

File: src/index.tsx [hunk 2]
```diff
@@ -1,52 +1,40 @@
...
-  return prepareFontFamily;
+  return fontFamily + '-' + FontWeightTranslations[fontWeight];
 }
 
 interface CustomProps extends TextProps {
-  component: React.FC<TextProps>;
-  computeFont: Function;
+  children?: ReactNode;
+  component?: React.ComponentType<TextProps>;
+  computeFont?: (
```
Comment (retyui) [inline src/index.tsx:15]: `component` + `computeFont` should be an options props

File: src/index.tsx [hunk 3]
```diff
@@ -1,52 +1,40 @@
...
-    prepareFontFamily = fontFamily + '-' + FontWeightTranslations[fontWeight];
-  }
-  return prepareFontFamily;
+  return fontFamily + '-' + FontWeightTranslations[fontWeight];
 }
 
 interface CustomProps extends TextProps {
-  component: React.FC<TextProps>;
-  computeFont: Function;
+  children?: ReactNode;
```
Comment (retyui) [inline src/index.tsx:13]: without children prop:

image

File: src/index.tsx [hunk 4]
```diff
@@ -1,52 +1,40 @@
...
 import FontWeightTranslations from './font-weight-translations';
 
 export function computeFontDefault(
-  fontFamily: string | undefined,
-  fontWeight: string = 'normal'
+  fontFamily: string,
+  fontWeight: TextStyle['fontWeight'] = 'normal'
 ): string {
-  let prepareFontFamily = '';
-  if (fontFamily) {
```
Comment (retyui) [inline src/index.tsx:14]: `fontFamily` is not nullable -  already checked in render function

File: src/font-weight-translations.tsx [hunk 5]
```diff
@@ -1,8 +1,8 @@
-interface Map {
-  [key: string]: string | undefined;
-}
+import type { TextStyle } from 'react-native';
 
-const FontWeightTranslations: Map = {
+type FontWeight = Exclude<TextStyle['fontWeight'], void>;
+
+const FontWeightTranslations: Record<FontWeight, string> = {
```
Comment (retyui) [inline src/font-weight-translations.tsx:5]: Strict type better than just a `string`

image

**Review by Vijay-mRoads — APPROVED** [review]
Thank you for your contributions. Merging the commits to master branch.

**Pull request merged**